### PR TITLE
[CatalogPromotion] refactor cp rule to be consistent with cp action

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         |master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
```
Comment (arti0090) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:3]: Should we add here a docblock?

**Review by arti0090 — APPROVED** [review]

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php (hunk 1 above)
Comment (SirDomin) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:3]: we dont have it everywhere, idk :/

Comment (arti0090) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:3]: @GSadee maybe - wdyt?

Comment (lchrusciel) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:3]: We should have them everywhere. Please add

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/form.xml [hunk 2]
```diff
@@ -160,6 +159,10 @@
             <argument>%sylius.model.avatar_image.class%</argument>
             <tag name="form.type" />
         </service>
+        <service id="sylius.form.type.variant_rule" class="Sylius\Bundle\PromotionBundle\Form\Type\CatalogRule\VariantRuleConfigurationType">
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/form.xml:162]:
```suggestion
        <service id="sylius.form.type.variant_rule" class="Sylius\Bundle\CoreBundle\Form\Type\CatalogRule\VariantRuleConfigurationType">
```

File: src/Sylius/Bundle/CoreBundle/Form/Extension/CatalogPromotionRuleTypeExtension.php [hunk 3]
```diff
@@ -5,32 +5,17 @@
 namespace Sylius\Bundle\CoreBundle\Form\Extension;
 
 use Sylius\Bundle\PromotionBundle\Form\Type\CatalogPromotionRuleType;
-use Sylius\Bundle\ResourceBundle\Form\Type\ResourceAutocompleteChoiceType;
+use Sylius\Bundle\PromotionBundle\Form\Type\CatalogRule\VariantRuleConfigurationType;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/CatalogPromotionRuleTypeExtension.php:8]:
```suggestion
use Sylius\Bundle\CoreBundle\Form\Type\CatalogRule\VariantRuleConfigurationType;
```
There should not be any mention about products/variants in promotion bundle

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace Sylius\Bundle\PromotionBundle\Form\Type\CatalogRule;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:5]:
```suggestion
namespace Sylius\Bundle\CoreBundle\Form\Type\CatalogRule;
```

File: src/Sylius/Bundle/CoreBundle/Validator/Constraints/CatalogPromotionRulesValidator.php [hunk 5]
```diff
@@ -40,13 +40,20 @@ public function validate($value, Constraint $constraint): void
...
 
-                $this->validateRuleConfiguration($rule->getConfiguration(), $constraint);
+            $this->validateRuleConfiguration($rule->getConfiguration(), $constraint);
         }
     }
 
     private function validateRuleConfiguration(array $configuration, CatalogPromotionRules $constraint): void
     {
-        foreach ($configuration as $variantCode) {
+
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Validator/Constraints/CatalogPromotionRulesValidator.php:49]:
```suggestion
```

File: src/Sylius/Bundle/CoreBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Form\Type\CatalogRule;
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:14]:
```suggestion
namespace Sylius\Bundle\CoreBundle\Form\Type\CatalogPromotionRule;
```

File: src/Sylius/Bundle/CoreBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+use Sylius\Bundle\ResourceBundle\Form\Type\ResourceAutocompleteChoiceType;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+
+final class VariantRuleConfigurationType extends AbstractType
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/Type/CatalogRule/VariantRuleConfigurationType.php:21]:
```suggestion
final class ForVariantsRuleConfigurationType extends AbstractType
```

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, @SirDomin! :1st_place_medal:

**Pull request merged**